instrumentation/net/http/otelhttp/handler.go

```diff
@@ -232,13 +233,14 @@ func (h *middleware) serveHTTP(w http.ResponseWriter, r *http.Request, next http
 		attributes = append(attributes, semconv.HTTPStatusCode(rww.statusCode))
 	}
 	o := metric.WithAttributes(attributes...)
-	h.requestBytesCounter.Add(ctx, bw.read, o)
-	h.responseBytesCounter.Add(ctx, rww.written, o)
+	ctxWithoutCancel := withoutCancel(ctx)
```

Comment:
Now that you've shared that [snippet from the context source](https://cs.opensource.google/go/go/+/refs/tags/go1.21.4:src/context/context.go;l=563-571) @pellared, is there a reason we can't use that directly? I.e.
```suggestion
	ctxWithoutCancel := context.WithoutCancel(ctx)
```

(and then remove the local implementation)?

Reply:
Because `context.WithoutCancel` was introduced in Go 1.21 and we also support Go 1.20.